modules/internal/ChapelStandard.chpl

```diff
@@ -26,51 +26,51 @@ module ChapelStandard {
-  public use ChapelTuple;
-  public use ChapelRange;
-  public use ChapelReduce;
-  public use ChapelSyncvar;
-  public use ChapelTaskDataHelp;
-  public use LocaleModel;
-  public use ChapelLocale;
-  public use ChapelPrivatization;
-  public use DefaultRectangular; // This might be able to go just after Atomics
-  public use LocalesArray;
-  public use ChapelArray;
-  public use ChapelDistribution;
-  public use ChapelAutoLocalAccess;
-  public use ChapelIO;
-  public use LocaleTree;
-  public use ChapelHashing;
-  public use DefaultAssociative;
-  public use DefaultSparse;
-  public use ChapelTaskID;
-  public use ChapelTaskTable;
-  public use MemTracking;
-  public use ChapelUtil;
-  public use Errors;
-  public use ChapelTaskData;
-  public use ChapelSerializedBroadcast;
-  public use ExportWrappers;
-  public use ChapelAutoAggregation;
+  public use CString as CString;
+  public use Bytes as Bytes;
+  public use String as String;
+  public use OwnedObject as OwnedObject;
+  public use SharedObject as SharedObject;
+  public use ChapelEnv as ChapelEnv;
+  public use ChapelBase as ChapelBase;
+  public use Atomics as Atomics;
+  public use NetworkAtomics as NetworkAtomics;
+  public use NetworkAtomicTypes as NetworkAtomicTypes;
+  public use AtomicsCommon as AtomicsCommon;
+  public use ChapelIteratorSupport as ChapelIteratorSupport;
+  public use ChapelThreads as ChapelThreads;
+  public use ChapelTuple as ChapelTuple;
+  public use ChapelRange as ChapelRange;
+  public use ChapelReduce as ChapelReduce;
+  public use ChapelSyncvar as ChapelSyncvar;
+  public use ChapelTaskDataHelp as ChapelTaskDataHelp;
+  public use LocaleModel as _; // let LocaleModel refer to the class
+  public use ChapelLocale as ChapelLocale;
+  public use ChapelPrivatization as ChapelPrivatization;
+  public use DefaultRectangular as DefaultRectangular; // This might be able to go just after Atomics
+  public use LocalesArray as LocalesArray;
+  public use ChapelArray as ChapelArray;
+  public use ChapelDistribution as ChapelDistribution;
+  public use ChapelAutoLocalAccess as ChapelAutoLocalAccess;
+  public use ChapelIO as ChapelIO;
+  public use LocaleTree as LocaleTree;
+  public use ChapelHashing as ChapelHashing;
+  public use DefaultAssociative as DefaultAssociative;
+  public use DefaultSparse as DefaultSparse;
+  public use ChapelTaskID as ChapelTaskID;
+  public use ChapelTaskTable as ChapelTaskTable;
+  public use MemTracking as MemTracking;
+  public use ChapelUtil as ChapelUtil;
+  public use Errors as Errors;
+  public use ChapelTaskData as ChapelTaskData;
+  public use ChapelSerializedBroadcast as ChapelSerializedBroadcast;
+  public use ExportWrappers as ExportWrappers;
+  public use ChapelAutoAggregation as ChapelAutoAggregation;
 
   // Standard modules.
-  public use Types;
-  public use Math;
-  public use VectorizingIterator;
+  public use Types as Types;
+  public use Math as Math;
+  public use VectorizingIterator as VectorizingIterator;
```

Comment:
Are these changes required for anything other than the standard modules?  I.e., we don't publicize (and in fact, would prefer to hide) the names of all internal modules from users.  So it seems as though we could potentially just change these final three (although, actually, I think there are a few standard modules in the first section now as well due to refactoring...  ChapelIO, Errors, ...).

Reply:
I agree that hiding the names of the internal modules is reasonable; however I don't think this PR should get into that. It leaves the names available just as they have been historically. I think it is likely that we have a few tests that rely on the names. Also, if we really want to hind the names, you shouldn't be able to `use` an internal module at all in user code.

Reply:
OK.  I may volunteer to take that on as a spare-time activity, as I'd like to lock down those names better (it's thematically related to other namespace lockdowns I was doing a year or two back).  But I'll wait for this to merge before taking it on.  Could potentially be done at the same time as the "compiler inserts standard module uses" effort if we thought that still had (any) value (I know you've explained that it won't help with some of the unfortunate `e` conflicts with `Math.e` cases).